symbol_exporter/ast_symbol_extractor.py

```diff
@@ -18,13 +18,16 @@ class SymbolType(str, Enum):
 
 
 class SymbolFinder(ast.NodeVisitor):
-    def __init__(self, module_name):
+    def __init__(self, module_name, filename=None):
```

Comment:
Why do we need filename? A python file is a module. 😄 

Is it the package we want to take into account?

Reply:
Yeah, I think we handle the mapping of the `__init__.py` back to the folder name, which I was hoping to handle with this but maybe we can translate it back to the proper file name. I can remove it for the time being and see how far we get.

Reply:
Yeah, we should discuss how to handle packages. I think it may be better to handle them at a higher level, leaving  `SymbolFinder` to parse ast nodes and and another (decorator class PackageSymbolFinder ?) that can take filenames and can walk a package hierarchy, passing the files into `SymbolFinder` and aggregating the results in a final symbols for the package.

PackageSymbolFinder can keep the dir hierarchy structure and call `SymbolFinder` for each file, knowing which is `__init__.py` and knowing the the nested level we are in, combining all calls into a final symbols object for the package.

I think separating the logic will help with maintainability and testing.

We can tackle regular packages first, we may also need to take into account namespace packages, [PEP-420](https://www.python.org/dev/peps/pep-0420/#differences-between-namespace-packages-and-regular-packages), where "Portions of namespace packages need not all come from the same directory structure".